src/coreComponents/physicsSolvers/fluidFlow/wells/WellControls.hpp

```diff
@@ -233,6 +233,12 @@ class WellControls : public dataRepository::Group
    */
   bool wellIsOpen( real64 const & currentTime ) const;
 
+  /**
+   * @brief Getter for the flag to disable crossflow in injectors
+   * @return the flag deciding whether crossflow is allowed or not
+   */
+  bool disableInjectorCrossflow() const { return m_disableInjectorCrossflow; }
```

Comment:
Is it relevant to get this flag for a `PRODUCER` well?
What happens in either case?

Reply:
No, for now, it is not relevant to get this flag for a producer. If a user tries to define the field `disableInjectorCrossflow` in the xml, I put a check in `WellControls.cpp` that will stop the simulation.

So inside the solver, the flag can be used to disable reservoir-to-well flow for injectors, but if someone calls it for a producer (despite the name), it will not do anything and will be ignored.

Reply:
How it would function if the injector becomes a producer (or vice versa) after certain time period, such as Pump-in/Flowback Tests?

Reply:
@francoishamon What I mean is that at some moment you 
```c++
WellControls::Type const wellType = wellControls.getType();
  bool const disableReservoirToWellFlow =
    ( wellType == WellControls::Type::INJECTOR ) && wellControls.disableInjectorCrossflow();
```
So this is a double check (that you've already made).

What I would find clearer, is to get a `disableInjectorCrossflow` feature more generic, like `disableCrossflow`.
If you want to prevent from using it with `PRODUCER`, then you can do i t like you do.
Reading
```c++
wellControls.isInjector() and wellControls.isCrossFlowDisabled()
```
is more meaningfull.

Moreover, I'm uncomfortable with the wording `disableInjectorCrossflow`.
When first reading it, I thought that calling this function would effectively disable the injector crossflow.
While it does not seem to be the case.
I would rename into `isCrossFlowDisabled` if it's actually the meaning you want to give to this function.

Also, I would add some `wellControls.isInjector()` and `wellControls.isProducer()` as well, it makes the reading more  fluid.

Reply:
I addressed the comments above: 

> What I would find clearer, is to get a `disableInjectorCrossflow` feature more generic, like `disableCrossflow`. If you want to prevent from using it with `PRODUCER`, then you can do i t like you do. Reading 
> ```c++
> wellControls.isInjector() and wellControls.isCrossFlowDisabled()
> ```

Done
 
> 
>  I would rename into `isCrossFlowDisabled` if it's actually the meaning you want to give to this function.

Done 

> Also, I would add some `wellControls.isInjector()` and `wellControls.isProducer()` as well, it makes the reading more fluid.

Done 

> EDIT: And maybe, at your convenience, rename `wellControls.wellIsOpen()` into `wellControls.isWellOpen()` for consistency.

Done

Reply:
> How it would function if the injector becomes a producer (or vice versa) after certain time period, such as Pump-in/Flowback Tests?

@jhuang2601 In the cases that you mention, disabling the crossflow would not be a good option to use. The new option implemented in this PR is more an expert option to use with caution in very specific cases. For now, I suggest using it when only 1) when the warning about crossflow pops up, and 2) when the presence of crossflow cripples nonlinear convergence. Otherwise, we should stick to the default behavior (i.e., crossflow enabled).